src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private int firstLeg;
+    private int secondLeg;
+    private int top;
+    private int bottom;
+
+    public IsoscelesTrapezoid(int firstLeg, int secondLeg, int top, int bottom, String color) {
+        super(color);
+        this.firstLeg = firstLeg;
+        this.secondLeg = secondLeg;
+        this.top = top;
+        this.bottom = bottom;
+    }
+
+    public int getFirstLeg() {
+        return firstLeg;
+    }
+
+    public int getSecondLeg() {
+        return secondLeg;
+    }
+
+    public int getTop() {
+        return top;
+    }
+
+    public int getBottom() {
+        return bottom;
+    }
+
+    @Override
+    public double getArea() {
+        if (getFirstLeg() == getSecondLeg() && getTop() > getBottom()){
```

Comment:
The condition 'getFirstLeg() == getSecondLeg() && getTop() > getBottom()' seems incorrect for determining if the trapezoid is isosceles. Typically, an isosceles trapezoid has non-parallel sides (legs) of equal length, and the top and bottom are the parallel sides. Consider revising this logic.